service/core/client/src/main/scala/com/lightbend/lagom/internal/client/ClientServiceCallInvoker.scala

```diff
@@ -48,26 +49,41 @@ private[lagom] abstract class ClientServiceCallInvoker[Request, Response](
       val requestSerializerStreamed = messageSerializerIsStreamed(requestSerializer)
       val responseSerializerStreamed = messageSerializerIsStreamed(responseSerializer)
 
-      val result: Future[(ResponseHeader, Response)] = (requestSerializerStreamed, responseSerializerStreamed) match {
-        case (false, false) =>
-          makeStrictCall(requestHeader, requestSerializer.asInstanceOf[MessageSerializer[Request, ByteString]],
-            responseSerializer.asInstanceOf[MessageSerializer[Response, ByteString]], request)
-        case (false, true) =>
-          makeStreamedResponseCall(requestHeader, requestSerializer.asInstanceOf[MessageSerializer[Request, ByteString]],
-            responseSerializer.asInstanceOf[MessageSerializer[Response, AkkaStreamsSource[ByteString, NotUsed]]], request)
-        case (true, false) =>
-          makeStreamedRequestCall(
-            requestHeader,
-            requestSerializer.asInstanceOf[MessageSerializer[Request, AkkaStreamsSource[ByteString, NotUsed]]],
-            responseSerializer.asInstanceOf[MessageSerializer[Response, ByteString]], request
-          )
-        case (true, true) =>
-          makeStreamedCall(
-            requestHeader,
-            requestSerializer.asInstanceOf[MessageSerializer[Request, AkkaStreamsSource[ByteString, NotUsed]]],
-            responseSerializer.asInstanceOf[MessageSerializer[Response, AkkaStreamsSource[ByteString, NotUsed]]], request
-          )
-      }
+      val result: Future[(ResponseHeader, Response)] =
+        (requestSerializerStreamed, responseSerializerStreamed) match {
+
+          case (false, false) =>
+            makeStrictCall(
+              headerFilterTransformClientRequest(headerFilter, requestHeader),
```

Comment:
I changed the code a little bit. Instead of adding another private method, I choose to convert the `RequestHeader` before calling the `makeXCall` methods. This is all internal/private methods and I didn't want to introduce another private method just to remove `RequestHeader` from the scope. 

Internally, we do other transformations (ie: for serializers), but there we are safe because we transform them to another type, `NegotiatedSerializer`, which protect us from passing the wrong instance further. Types to the rescue!